openforcefield/tests/test_toolkits.py

```diff
@@ -657,77 +659,221 @@ def test_generate_conformers(self):
-        molecule.generate_conformers(toolkit_registry=toolkit_wrapper)
-        # Ensure that an exception is raised if an invalid charge model is passed in
-        with pytest.raises(Exception) as excinfo:
-            charge_model = 'notARealChargeModel'
-            molecule.compute_partial_charges(toolkit_registry=toolkit_wrapper, charge_model=charge_model)
-
-        # TODO: Test all supported charge models
-        # Note: "amber" and "amberff94" only work for a subset of residue types, so we'll need to find testing data for
-        # those
-        # charge_model = [,'amber','amberff94']
-        # TODO: 'mmff' and 'mmff94' often assign charges of 0, presumably if the molecule is unrecognized.
-        # charge_model = ['mmff', 'mmff94']
-        for charge_model in ['noop', 'am1bcc', 'am1bccnosymspt', 'am1bccelf10']:
-            with pytest.raises(NotImplementedError) as excinfo:
-                molecule.compute_partial_charges(toolkit_registry=toolkit_wrapper)  # , charge_model=charge_model)
-                charge_sum = 0 * unit.elementary_charge
-                for pc in molecule._partial_charges:
-                    charge_sum += pc
-                assert charge_sum < 0.001 * unit.elementary_charge
-
-        # For now, just test AM1-BCC while the SMIRNOFF spec for other charge models gets worked out
-        molecule.compute_partial_charges_am1bcc(toolkit_registry=toolkit_wrapper)  # , charge_model=charge_model)
+        molecule.generate_conformers(rms_cutoff=1*unit.angstrom, n_conformers=100)
+        assert molecule.n_conformers > 1
+        assert not(molecule.conformers[0] == (0.*unit.angstrom)).all()
+
+        # Ensure rms_cutoff kwarg is working
+        molecule2 = toolkit_wrapper.from_smiles(smiles)
+        molecule2.generate_conformers(rms_cutoff=0.1*unit.angstrom, n_conformers=100)
+        assert molecule2.n_conformers > molecule.n_conformers
+
+        # Ensure n_conformers kwarg is working
+        molecule2 = toolkit_wrapper.from_smiles(smiles)
+        molecule2.generate_conformers(rms_cutoff=0.1*unit.angstrom, n_conformers=10)
+        assert molecule2.n_conformers == 10
+
+    @pytest.mark.skipif(not OpenEyeToolkitWrapper.is_available(), reason='OpenEye Toolkit not available')
+    def test_compute_partial_charges_am1bcc(self):
+        """Test OpenEyeToolkitWrapper compute_partial_charges_am1bcc()"""
+        toolkit_registry = ToolkitRegistry(toolkit_precedence=[OpenEyeToolkitWrapper])
+        molecule = create_ethanol()
+        molecule.compute_partial_charges_am1bcc(toolkit_registry=toolkit_registry)  # , charge_model=charge_model)
         charge_sum = 0 * unit.elementary_charge
+        abs_charge_sum = 0 * unit.elementary_charge
         for pc in molecule._partial_charges:
             charge_sum += pc
-        assert charge_sum < 0.001 * unit.elementary_charge
+            abs_charge_sum += abs(pc)
+        assert abs(charge_sum) < 0.005 * unit.elementary_charge
```

Comment:
I had to increase the threshold because AmberTools only records to 3 digits of precision in charges.